templates/CRM/Report/Form/Layout/Overlay.tpl

```diff
@@ -102,6 +102,10 @@
                             {/if}
                         {elseif $header.type eq 1024}
                             <span class="nowrap">{$row.$field|crmMoney}</span>
+                        {elseif is_array($row)}
+                            {foreach from=$row.$field item=fieldrow key=fieldid}
+                                <div class="crm-report-{$field}-row-{$fieldid}">{$fieldrow}</div>
```

Comment:
Cool it seems like it's almost there. A couple tuneup notes:
* $row is always an array - you want to check `is_array($row.$field)`
  * Related to that, smarty seems to have some kind of autocoercion - if $row.$field is a scalar you can still loop over it and it will come out fine, but it's unintentional. For example run the contribution detail report and you can see it's applying this loop/div.
* I think the `is_array` check should be at the top of the `if` block, before Subtotal and change Subtotal to elseif, because otherwise it's technically a "Notice: array to string conversion" comparison that happens for that Subtotal check for the logging report.

Reply:
You are absolutely right! I missed seeing that actually the `$row.$field` is the one that we're looking for!

As on the where to check if the field is an array, I'm moving it on the next commit, it does make sense: avoid the array to string comparison error before it actually happens.

In regards on how smarty works itself with scalar values, i wasn't able to reproduce it on the contrib detail report but i do get what you say.